chromium_src/chrome/browser/ui/browser_navigator.cc

```diff
@@ -21,23 +24,35 @@ void UpdateBraveScheme(NavigateParams* params) {
   }
 }
 
-bool IsHostAllowedInIncognitoBraveImpl(const std::string_view host) {
-  if (host == kWalletPageHost || host == kWalletPanelHost ||
-      host == kRewardsPageHost || host == chrome::kChromeUISyncInternalsHost ||
+bool IsURLAllowedInIncognitoBraveImpl(
+    const GURL& url,
+    content::BrowserContext* browser_context) {
+  std::string scheme = url.scheme();
+  std::string_view host = url.host_piece();
+  if (scheme != content::kChromeUIScheme) {
+    return true;
+  }
```

Comment:
Should that allow all `brave://` for incognito?

Reply:
It shouldn't - for example, you can't load brave://extensions in a private window on this branch.

The intention is to keep behavior exactly the same as before, except for the wallet, which now depends on the pref value.

For context, previously, we patched IsHostAllowedInIncognito, making modifications for our URLs. However, in this change I patch IsURLAllowedInIncognito instead because that function has access to the browser context needed to fetch the pref value.

This part
```
  if (scheme != content::kChromeUIScheme) {
    return true;
  }
```

Is not actually new code, but existing code taken from IsHostAllowedInIncognito, that is put here so we mimic the same behavior as before.